### PR TITLE
BUG: Not all viewers have get_limits

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to fix broken pan/zoom on Plot Options histogram.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-4997)

### CODE REVIEW

File: jdaviz/core/tools.py [hunk 1]
```diff
@@ -40,7 +40,8 @@ class _BaseZoomHistory:
     # Mixin for custom zoom tools to be able to save their previous zoom state
     # which is then used by the PrevZoom tool
     def save_prev_zoom(self):
-        self.viewer._prev_limits = self.viewer.get_limits()
+        if hasattr(self.viewer, "get_limits"):
+            self.viewer._prev_limits = self.viewer.get_limits()
```
Comment (rosteen) [inline jdaviz/core/tools.py:44]: So what happens if you press the `previous zoom` button in these viewers? Nothing?

Comment (pllim) [reply jdaviz/core/tools.py:44]: For this particular bug, it is the Plot Options histogram, so there is no previous zoom button. It just somehow inherited that method.

**Comment by rosteen** [comment]
The histogram viewer actually does have previous zoom:

Screenshot 2024-12-17 at 12 44 04 PM

And now clicking it gives this error instead:

```
File [~/projects/jdaviz/jdaviz/core/tools.py:166](http://localhost:8888/lab/tree/testing/~/projects/jdaviz/jdaviz/core/tools.py#line=165), in PrevZoom.activate(self)
    165 def activate(self):
--> 166     if self.viewer._prev_limits is None:
    167         return
    168     prev_limits = self.viewer._prev_limits

AttributeError: 'BqplotHistogramView' object has no attribute '_prev_limits'
```

**Comment by pllim** [comment]
Ooops... Should it???

**Comment by rosteen** [comment]
@kecnry thoughts? Probably not particularly needed here, it would be nice to have but maybe if we can't support it here we just remove it.

**Comment by kecnry** [comment]
Is there another way to get the limits (or implement get_limits)?

**Comment by pllim** [comment]
p.s. I knew it was too easy! I'll look more.

**Comment by pllim** [comment]
I think `get_limits` is something we invented in Jdaviz. It is not available for a viewer class we used directly from glue-jupyter, e.g., https://github.com/glue-viz/glue-jupyter/blob/main/glue_jupyter/bqplot/histogram/viewer.py

https://github.com/spacetelescope/jdaviz/blob/629fa623b841cd6564000de3605a5aa74520743b/jdaviz/configs/default/plugins/viewers.py#L170

**Comment by pllim** [comment]
bisect blame the breakage on https://github.com/spacetelescope/jdaviz/pull/3262 . Pretty sure it broke between v4.0.0 and v4.0.1

**Comment by pllim** [comment]
Thanks for the review. I basically reverted the offending change that I added in #3249 . Should work now.

**Review by rosteen — APPROVED** [review]
Thanks!

**Review by kecnry — APPROVED** [review]

**Pull request merged**